### PR TITLE
Webpack glslify

### PR DESCRIPTION
**Pull request by adnan-wahab**

### CODE REVIEW

**Comment by adnan-wahab** [comment]
@alexanderGugel @michaelobriena @redwoodfavorite @FarhadG

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "lint": "eslint --ignore-path .gitignore .",
     "build": "npm run build-debug && npm run build-standalone",
     "build-debug": "browserify index.js -d --standalone famous > dist/famous.js",
-    "build-min": "browserify index.js --standalone famous | uglifyjs --screw-ie8 -m -c dead_code,sequences,conditionals,booleans,unused,if_return,join_vars,drop_debugger > dist/famous.min.js"
+    "build-min": "browserify index.js --standalone famous | uglifyjs --screw-ie8 -m -c dead_code,sequences,conditionals,booleans,unused,if_return,join_vars,drop_debugger > dist/famous.min.js",
+    "build-shaders": "node ./scripts/bundle-shaders.js"
```
Comment (alexanderGugel) [inline package.json]: Maybe we should ass this as a `prepublish` script @TheAlphaNerd

File: package.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
     "lint": "eslint --ignore-path .gitignore .",
     "build": "npm run build-debug && npm run build-standalone",
```
Comment (alexanderGugel) [inline package.json]: Should be

```
     "build": "npm run build-shaders && npm run build-debug && npm run build-standalone",  
```

File: scripts/build-shaders.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+function writeFile(src, target) {
+    fs.writeFile(path.resolve(__dirname, '..', 'webgl-shaders', target),
+                 makeModule(src),
+                 function(err) { if (err) throw err; }
+                );
+}
+
+function makeModule(file) {
+    return '\'use strict\'; module.exports = ' + file + ';';
```
Comment (alexanderGugel) [inline scripts/build-shaders.js]: Can we add `\n` at the end of the file? That way git doesn't complain....

**Comment by TheAlphaNerd** [comment]
There are a bunch of issues here.

1) This should be designed as a prepublish script
2) It should be included in the build pipeline
3) Build Assets should not be checked into git
4) We need a reasonable workflow for development (this includes both working on engine, and engine not from npm inside of a project)

I think that this might be complex enough to be worth breaking out into it's own module. If it were its own module engine could completely not care about the build step, and the logic for supporting dev could be offloaded to the other module

**Comment by alexanderGugel** [comment]
1) This is literally a one liner. Added comment above. Good point.
2) Why can't we have this pipeline (which is just a single script) in the repo itself?
3) Kind of agree, but this is the most pragmatic way how we can solve this at the moment. Anything else introduces additional complexity that might not worth the effort at the moment. Especially since the shaders aren't something that changes on a day-to-day basis (as opposed to actual builds).
4) I think this is currently the most reasonable and pragmatic way.

This is literally a single script. I understand that you want to break this out, but I think we made bad experiences with extreme modularization (I would argue that breaking out such a single script into a separate module is). I understand your concerns and agree in some way, but I don't think there is anything wrong with the proposed solution until we have a concrete proposal for this pipeline.

**Pull request closed without merging**